### PR TITLE
Add larcoh readme

### PR DESCRIPTION
**Pull request by KatalinaBobowik**
Add additional README for running large cohort pipeline. Specifically, this details the input parameters for each stage and what they do. This will enable users to know what they would like to change, and what options they have.

I made a `Notes on what we need` section, which highlights most of the things I think will help with this README. For a lot of the thresholds used, I'm assuming values were pulled from gnomAD QC, but if a link could be provided that would be fantastic, as there are different thresholds for different gnomAD versions. Once these are filled out, I can finish up the `To do` section.

@vladsavelyev There are quite a few sections, particularly in the `Configurable inputs` section at the bottom. It'd be nice to have all of these inputs filled out, as well as which config file they are applicable to. 

Note: I highlighted all of my questions/bits I need help with using the `==text==` syntax (which I couldn't get to work ☹️), but I'll tag the relevant people for those lines instead. If anyone feel GitHub issues would be better or make this easier to break down, I'm happy to do that, too!

### CODE REVIEW

File: docs/large_cohort_pipeline_readme.md [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+3. FrequenciesStage: Runs the [VDS combiner](https://hail.is/docs/0.2/vds/hail.vds.combiner.VariantDatasetCombiner.html#hail.vds.combiner.VariantDatasetCombiner "") with GVCFs as an input to produce a VariantDataset (VDS), then performs sample-level QC (pedigree, sex and ancestry inference)
+4. LoadVqsr Stage: Converts the VDS into a sites-only VCF-ready table, then performs allele-specific VQSR variant QC.
+
+Graphically, the pipeline has the following workflow:
+
+![uml](large_cohort.png)
+
+## Running the pipeline
+To run, the pipeline takes in at least two config files: an exome/genome config file, specifying whether samples are whole genome or whole exome sequences (e.g., `configs/genome.toml`), and a config file specifying parameters specific to the dataset. This second config file **must** be named after the dataset + access level, e.g., `configs/bioheart-test.toml` . By default, a third config file,  `configs/defaults/large_cohort.toml` , is also loaded into the pipeline. This config file does (==x, y, z - how is this different to the dataset + access level config file?==). Any additional config files can be provided by adding  `--config` to the analysis runner command.
```
Comment (KatalinaBobowik) [inline docs/large_cohort_pipeline_readme.md:14]: @vladsavelyev just curious how the dataset + access level config file is differnet to the larcoh file? I'm not really sure which one feeds into which parts of the script

File: docs/large_cohort_pipeline_readme.md [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+<br><ins>Configurable inputs:</ins>
+	* [workflow].vds_version
+	* [hail].dataproc.combiner_autoscaling_policy
+	* get_config().get('combiner', {})
+	* [workflow].[sequencing_type]
+	* [workflow].check_inputs
+	* [workflow].skip_samples_with_missing_input
+
+3. SampleQC: This step takes an input VDS table and performs the following steps: 1. removes centromeres and telomeres 2. filters to autosomes and 3. computes sample quality metrics about the VDS (==what specifically are these? Can’t find in the `vds.sample_qc` docs==). Sex is then imputed on the VDS and the following soft filters added: samples with ambiguous sex assignments, low-coverage samples, and extreme raw bi-allelic sample QC outliers. Sample-QC based filters are calculated according to the thresholds specified in the `large_cohort.sample_qc_cutoffs` section of the `configs/defaults/large_cohort.toml`. The output of this step is written to `gs://cpg-{dataset}-{access-level}/large-cohort/v0-1/sample_qc.ht`.
```
Comment (KatalinaBobowik) [inline docs/large_cohort_pipeline_readme.md:99]: @vladsavelyev @lgruen would you happen to have any extra info of what sample quality metrics are used here? I couldn't find it in the docs

File: docs/large_cohort_pipeline_readme.md [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+<ins>Configurable inputs:</ins>
+	* [workflow].[sequencing_type]
+	* [workflow].[sequencing_type]
+	* [large_cohort].[sample_qc_cutoffs]
+
+4. DenseSubset: This step filters a sparse VDS to a set of predetermined QC sites and returns a dense (no filtered entries) MatrixTable with split multiallelics. The markers for the subset are read from the `references.gnomad.predetermined_qc_variants` Hail table specified in the TOML config (==which config?==). The table is suitable for both exomes and genomes, so that a mixture of different sequencing types can be processed together. The resulting subset is written to `gs://cpg-{dataset}-{access-level}/large-cohort/v0-1/dense-subset.mt`.
```
Comment (KatalinaBobowik) [inline docs/large_cohort_pipeline_readme.md:105]: @vladsavelyev Does the `references.gnomad.predetermined_qc_variants` parameter come from the default larcoh.toml, or the one specified by the user?

File: docs/large_cohort_pipeline_readme.md [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+* Where is `get_config().get()` being pulled from?
+
+## To do:
+- [ ] ensure all correct default values are added in the configurable inputs parameters field
+- [ ] make all descriptions in configurable inputs easy to understand (some of these are still hard to follow)
+- [ ] redistribute settoings in `unsure` configurable inputs section to corresponding toml files
+
+## Configurable inputs
+
+### unsure
```
Comment (KatalinaBobowik) [inline docs/large_cohort_pipeline_readme.md:181]: @vladsavelyev @vivbak for all the parameters below, I'm unsure of what config file they're being pulled from. I also don't have all the available info for some of them and not sure of what they're doing. Is it possible to get more info in the `description` section for the ones with `Need info`?

File: docs/large_cohort_pipeline_readme.md (hunk 1 above)
Comment (vladsavelyev) [reply docs/large_cohort_pipeline_readme.md:14]: There are no limitations on names and number of configs. You can create only one config file and call it `myconfig.toml`, add all required fields there including `sequencing_type`. Or you can split it up, it's just the matter of convenience. E.g. current [prophecy](https://github.com/populationgenomics/production-pipelines/blob/main/configs/prophecy.toml) and [bioheart](https://github.com/populationgenomics/production-pipelines/blob/main/configs/bioheart-test.toml) configs contain all parameters needed to run the large-cohort pipeline, including `sequencing_type = 'genome'`.

Comment (vladsavelyev) [reply docs/large_cohort_pipeline_readme.md:14]: `configs/defaults/large_cohort.toml` contains the defaults, applicable to the workflow in general, regardless of a dataset. You don't need to worry about it unless you want some reference to override certain parameters in your run-specific configs, like e.g. `bioheart.toml`

File: docs/large_cohort_pipeline_readme.md [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+--output-dir "bioheart" --access-level test \
+--config configs/genome.toml --config configs/bioheart-test.toml \
+--image australia-southeast1-docker.pkg.dev/cpg-common/images/cpg_workflows:latest main.py large_cohort
+```
+
+## Detailed stage descriptions
+The detailed description for each stage is as follows:
+### CramMultiQC Stage
+
+1.  Alignment: the first step is to align the given input files. By default, DRAGMAP version 1.3.0 is used (==where is this actually fed into the script from the config file?==). This then outputs a CRAM file with a corresponding index ==(what does that mean?)==.
```
[the remark this answers is not in the document]
Comment (vladsavelyev) [reply docs/large_cohort_pipeline_readme.md:29]: The version of the aligner matches the tag of the corresponding image used for the alignment job, and can be overridden in the config `images` section (default is https://github.com/populationgenomics/production-pipelines/blob/main/cpg_workflows/defaults.toml#L2)

File: docs/large_cohort_pipeline_readme.md (hunk 2 above)
Comment (vladsavelyev) [reply docs/large_cohort_pipeline_readme.md:99]: It's configured through toml as well: https://github.com/populationgenomics/production-pipelines/blob/main/configs/defaults/large_cohort.toml#L30-L36

The thresholds are borrowed from gnomAD QC, e.g. https://macarthurlab.org/2017/02/27/the-genome-aggregation-database-gnomad/

File: docs/large_cohort_pipeline_readme.md (hunk 3 above)
Comment (vladsavelyev) [reply docs/large_cohort_pipeline_readme.md:105]: It's specified in https://github.com/populationgenomics/production-pipelines/blob/main/configs/defaults/large_cohort.toml, as you can see. But as every other config entry, it can be overridden by user

File: docs/large_cohort_pipeline_readme.md (hunk 1 above)
Comment (KatalinaBobowik) [reply docs/large_cohort_pipeline_readme.md:14]: got it, thank you!

File: docs/large_cohort_pipeline_readme.md (hunk 3 above)
Comment (KatalinaBobowik) [reply docs/large_cohort_pipeline_readme.md:105]: perfect, thank you!

File: docs/large_cohort_pipeline_readme.md (hunk 2 above)
Comment (KatalinaBobowik) [reply docs/large_cohort_pipeline_readme.md]: Thanks @vladsavelyev . I can't find all the QC metrics in that link. Is it possible to give another source where you found those, and what gnomAD version it corresponds to?

File: docs/large_cohort_pipeline_readme.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (KatalinaBobowik) [reply docs/large_cohort_pipeline_readme.md:29]: Great, so would it be correct to say something along the lines of "By default, DRAGMAP is used, with the version matching the tag of the corresponding image used for the alignment job (specified in. [images] within the config file)"?

File: docs/large_cohort_pipeline_readme.md [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+* Why does `[workflow].sequencing_type` need to be specified in a separate exome/genome config file? Can this not be added in one of the other toml files?
+* Could the VQSR section in the README and `large_cohort.toml` be described better? I have no idea what these are doing, and why these inputs are chosen. Also, is this only to be specified in the `large_cohort.toml`?
+* Where is `get_config().get()` being pulled from?
+
+## To do:
+- [ ] ensure all correct default values are added in the configurable inputs parameters field
+- [ ] make all descriptions in configurable inputs easy to understand (some of these are still hard to follow)
+- [x] redistribute settings in `unsure` configurable inputs section to corresponding toml files
+
+## Configurable inputs
```
Comment (KatalinaBobowik) [inline docs/large_cohort_pipeline_readme.md:179]: @vladsavelyev @vivbak I'm having some difficulty filling out the parameters and descriptions to a lot of the parameters below. Could you fill in any of the blanks/highlighted below?

Comment (vladsavelyev) [reply docs/large_cohort_pipeline_readme.md:179]: Those are purely internal parameters, just for convenience of code debugging. I don't really think we need to document them in the large cohort pipeline documentation.

Comment (KatalinaBobowik) [reply docs/large_cohort_pipeline_readme.md:179]: are these not inputs you enter into the config that affect the results? It'd be good to know what they do so the user can decide what options are best for their workflow

Comment (vladsavelyev) [reply docs/large_cohort_pipeline_readme.md:179]: They are made for developers, and they won't affect the results

Comment (vladsavelyev) [reply docs/large_cohort_pipeline_readme.md:179]: But they are already documented in the large-cohort TOML https://github.com/populationgenomics/production-pipelines/blob/main/configs/defaults/large_cohort.toml#L21-L37 and in the existing README https://github.com/populationgenomics/production-pipelines/blob/main/README.md#sample-qc

I can add more details, but they are pretty self-explanatory and can be easily followed in the code, in my opinion

Comment (vladsavelyev) [reply docs/large_cohort_pipeline_readme.md:179]: Added more details on that into my documentation update PR :) https://github.com/populationgenomics/production-pipelines/pull/247
